### DISCUSSION TITLE
After Gitea upgrade to v1.20.1 renovate bot stopped working with FATAL: Authentication failure

### DISCUSSION THREAD

**Discussion by CodeShakingSheep** [renovatebot/renovate#23606] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.131.0

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Describe the problem

After I upgraded my Gitea from I believe 1.18 to 1.20.1 renovate bot stopped working because of authentication failure (see log below). I checked the Gitea PAT configured as renovate token which was still present in Gitea after the update. I also checked that it was correctly set when running renovate.

As renovate bot runs successfully when I use a PAT with all permissions granted, I believe this must be related to the token permissions.

The PAT "Renovate" I have used since I installed Renovate which has always worked until the Gitea upgrade has these permissions:
- write:package
- write:repository

[Here](https://docs.renovatebot.com/modules/platform/gitea/) it's stated that the PAT should have the following permissions:

- repo
- read:user and read:email
- If you use Gitea packages, add the read:packages scope.

So, I added another PAT "Renovate2" with the following permissions:
- read:package
- write:repository
- read:user
It failed with the same authentication error.

So, I have three questions here:

1. What are the permissions required for running renovate?  
  As my PAT with all permissions can be run successfully with renovate and the other PATs with dedicated Renovate permissions not, I wonder what permissions are missing to them.
2. What's the `read:email` permission called in the current Gitea release?
  I couldn't find this permission anywhere when setting PAT permissions. So, may be the name changed?
3. Is it safe to upgrade to renovate 36.25.4?  
  I read that version 36 has some breaking changes and that Node v19 is no longer supported. However, I'm running node 18.17.0 on my server.

Thank you for all, renovate is awesome! :)

### Relevant debug logs

<details><summary>Logs</summary>

```
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","config":{"platform":"gitea","endpoint":"[REDACTED_HOSTNAME]/api/v1/","gitAuthor":"Renovate Bot <renovate-bot@[REDACTED_HOSTNAME]>","username":"renovate-bot","autodiscover":tr>
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","config":{"logFile":"renovate.log","token":"***********"},"msg":"CLI config","time":"2023-07-29T10:56:09.694Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","config":{"hostRules":[]},"msg":"Env config","time":"2023-07-29T10:56:09.694Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","config":{"platform":"gitea","endpoint":"[REDACTED_HOSTNAME]/api/v1/","gitAuthor":"Renovate Bot <renovate-bot@[REDACTED_HOSTNAME]>","username":"renovate-bot","autodiscover":tr>
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","msg":"Adding trailing slash to endpoint","time":"2023-07-29T10:56:09.695Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","msg":"Found valid git version: 2.33.0","time":"2023-07-29T10:56:09.717Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","msg":"GET [REDACTED_HOSTNAME]/api/v1/version = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=403 retryCount=0, duration=19)","time":"2023-07-29T10:56:09.836Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","err":{"name":"HTTPError","code":"ERR_NON_2XX_3XX_RESPONSE","timings":{"start":1690628169815,"socket":1690628169816,"lookup":1690628169817,"connect":1690628169818,"secureConnect":16906281>
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":60,"logContext":"zhfMFQH1jIzcWy2dlmB8z","msg":"Authentication failure","time":"2023-07-29T10:56:09.838Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":20,"logContext":"zhfMFQH1jIzcWy2dlmB8z","msg":"Renovate exiting","time":"2023-07-29T10:56:09.839Z","v":0}
{"name":"renovate","hostname":"[REDACTED_HOSTNAME]","pid":1415707,"level":30,"logContext":"zhfMFQH1jIzcWy2dlmB8z","loggerErrors":[{"name":"renovate","level":60,"logContext":"zhfMFQH1jIzcWy2dlmB8z","msg":"Authentication failure"}],"msg":"Renovate is exiting with a non-zero code due to the following lo>
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
please provide the http stats debug log message, there you can see the failed requests

**Reply by CodeShakingSheep** [reply]
I updated the logs. The failed request seems to be `[REDACTED_HOSTNAME]/api/v1/version` which is interesting. I tested this endpoint with postman and if I query it with no authentication at all it returns HTTP 200
```json
{
    "version": "1.20.1"
}
```

However, if I query it with my "Renovate 2" PAT token it returns HTTP 403 Forbidden
```json
{
    "message": "token does not have at least one of required scope(s): [read:misc]",
    "url": "[REDACTED_HOSTNAME]/api/swagger"
}
```

So, I guess this is the answer to my first question that the PAT also requires [read:misc] permission. This should be updated in the [docs](https://docs.renovatebot.com/modules/platform/gitea/)  However, this seems like an inconsistent API design from Gitea to allow unauthorized requests to the endpoint but not a request with a PAT. I will report an issue about this in gitea repo.

Can you still elaborate on my other questions? Thanks.

**Reply by viceice** [reply]
yes, you need misc scope.

updating to v36 is safe, we still support node v18 lts. in general renovate always supports latest lts and non-eol node versions. we drop v19, because it's EOL.

**Reply by viceice** [reply]
the email scope seems to no longer exists on Gitea v1.20+

**Reply by CodeShakingSheep** [reply]
Alright, that's good to know regarding the update to v36. Thanks.

When creating a PAT with misc:read permission I ran into a further issue: The issues endpoint `[REDACTED_HOSTNAME]/api/v1/repos/[REDACTED_ORGANIZATION]/[REDACTED_REPO_NAME]/issues?state=all&type=issues` returns

```json
{
    "message": "token does not have at least one of required scope(s): [read:issue]",
    "url": "[REDACTED_HOSTNAME]/api/swagger"
}
```

So, apparently scope [read:issue] is required and probably scope [write:issue] too for editing the dependency dashboard issues. Can you confirm that?

Would be awesome if the [docs](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/gitea/index.md) could be updated with the additional scopes soon.

**Reply by viceice** [reply]
feel free to open a PR to update the docs. 🤗

**Reply by CodeShakingSheep** [reply]
> feel free to open a PR to update the docs. 🤗

Done 🙌 https://github.com/renovatebot/renovate/pull/23615

**Comment by viceice** [comment, marked as answer]
https://github.com/go-gitea/gitea/issues/26035
